File: Testing/Code for Testing/Early Basic Syntax Testing/packed/test_packed.js
// @packed - no padding (packed struct)
// bit field: nibble_hi = 4 bits
// bit field: nibble_lo = 4 bits
function createPackedData() {
    return { nibble_hi: 0, nibble_lo: 0, byte_val: 0 };
}


function main() {
    let d = createPackedData();
    d.nibble_hi = 15;
    d.nibble_lo = 9;
    d.byte_val = 200;
    console.log(d.nibble_hi);
    console.log(d.nibble_lo);
    console.log(d.byte_val);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/packed/test_packed.py
from dataclasses import dataclass, field

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

# @packed - no padding (packed struct)
@dataclass
class PackedData:
    nibble_hi: int = 0
    nibble_lo: int = 0
    byte_val: int = 0


def main():
    d = PackedData(nibble_hi=15, nibble_lo=9, byte_val=200)
    print(d.nibble_hi)
    print(d.nibble_lo)
    print(d.byte_val)
    return 0


if __name__ == "__main__":
    main()
